plugins/api/src/data_object_finalize.cpp

```diff
@@ -357,6 +397,17 @@ namespace
             return SYS_CONFIG_FILE_ERR;
         }
 
+        const auto data_id = std::stoll(replicas.front().at("before").at("data_id").get<std::string>());
+        const auto entity_type = ic::entity_type_map.at("data_object");
+        if (!admin_operation && !ic::user_has_permission_to_modify_metadata(*_comm, db_conn, data_id, entity_type)) {
```

Comment:
Mmm... when I read the query and access level in that function, it seemed to me that it was sufficient, but I think you're right that it would be more correct to implement a separate check for modifying replicas as they use a separate value for that permission. I'll do that.

Reply:
Through offline discussion, decided that `user_has_permission_to_modify_metadata` should be changed to be `user_has_permission_to_modify_entity` and can be used generically as done here.